lib/shelley/src/Cardano/Wallet/Shelley.hs

```diff
@@ -354,21 +353,22 @@ serveWallet
                 poolsDbTracer
                 (Pool.defaultFilePath <$> dir)
                 (timeInterpreter nl)
-                $ \db -> do
-            let spl = newStakePoolLayer nl db
+                $ \db@DBLayer{..} -> do
+
             void $ forkFinally (monitorStakePools tr gp nl db) onExit
-            fetch <- fetchFromRemote trFetch fetchStrategies
-                <$> newManager defaultManagerSettings
-            void $ forkFinally (monitorMetadata tr gp fetch db) onExit
+            spl <- newStakePoolLayer nl db (\mvar -> forkFinally'
+                (monitorMetadata tr gp db mvar)
+                onExit)
+            putSettings spl (fromMaybe defaultSettings settings)
             action spl
       where
         tr = contramap (MsgFromWorker mempty) poolsEngineTracer
-        trFetch = contramap MsgFetchPoolMetadata tr
         onExit = defaultWorkerAfter poolsEngineTracer
-        fetchStrategies = pure $ maybe
-            identityUrlBuilder
-            registryUrlBuilder
-            smashURL
+
+        forkFinally' :: IO a -> (Either SomeException a -> IO ()) -> IO (Async ())
+        forkFinally' action' and_then =
+          mask $ \restore ->
+            async $ try (restore action') >>= and_then
```

Comment:
Why is this needed exactly? What's wrong with `forkIO` in this context?

Reply:
Doesn't give us an async handle. That means we'd have to do extra work just to make sure the thread terminated (e.g. passing yet another MVar).

Reply:
It does.